operator/v1/types_ingress.go

```diff
@@ -1346,6 +1385,59 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// headersManipulation specifies options for set/replace/append/delete headers.
+	HeadersManipulation IngressControllerHeadersManipulation `json:"headersManipulation,omitempty"`
+}
+
+// IngressControllerHeadersManipulation specifies options for set/replace/append/delete headers.
+type IngressControllerHeadersManipulation struct {
+	Response []IngressControllerSetReplaceAppendDeleteHTTPHeaders `json:"response,omitempty"`
+}
+
+// IngressControllerSetReplaceAppendDeleteHTTPHeaders specifies fields to mention the headers and the action you want to perform on them.
+// +union
+type IngressControllerSetReplaceAppendDeleteHTTPHeaders struct {
+	// action lets you specify headers for Set or Delete.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Set;Delete
+	// +kubebuilder:validation:Required
+	Action IngressControllerSetHTTPHeaderAction `json:"action,omitempty"`
+
+	// name specifies a header name to be set(replace)/append/delete.  Its value must be a valid HTTP header.
+	// name as defined in RFC 2616 section 4.2.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	// +optional
+	Name *string `json:"name"`
+
+	// Set defines HTTP headers that should be set or replaced or appended.
+	// If this field is empty, no headers are added.
+	//
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set(replace)/append for TLS passthrough
+	// connections.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Set *IngressControllerSetHTTPHeader `json:"set,omitempty"`
+
+	// delete defines HTTP headers that should be deleted.
+	// If this field is empty, no headers are deleted.
+	//
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set(replace)/append for TLS passthrough
+	// connections.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Delete *IngressControllerDeleteHTTPHeader `json:"delete,omitempty"`
```

Comment:
dont think we need `IngressControllerDeleteHTTPHeader` type if it should a an empty struct.

Reply:
It was discussed in the Enhancement Proposal to have it. https://github.com/openshift/enhancements/pull/1296#discussion_r1061504552

Reply:
https://github.com/openshift/enhancements/pull/1296#discussion_r1064670236

Reply:
It is more clearer to include it.

Reply:
We do have several union member types that are empty structs in the IngressController API, namely `AWSNetworkLoadBalancerParameters` and `ContainerLoggingDestinationParameters`.  We also have union member types that were initially empty struct types, which we later expanded: `AWSClassicLoadBalancerParameters`, `HostNetworkStrategy`, `PrivateStrategy`, and `NodePortStrategy`.

If we make this an empty union member (as opposed to a union member with an empty struct type) now, and we later need to add a struct with one or more fields for the union member, would that be a breaking change?

Reply:
@JoelSpeed, do you know the answer to my question, or do we need to do some investigation?

Reply:
> If we make this an empty union member (as opposed to a union member with an empty struct type) now, and we later need to add a struct with one or more fields for the union member, would that be a breaking change?

You may convert an empty union member to a non-empty union member so long as there is nothing required in the configuration of the member, so the struct and all fields within it, must be optional